CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 2.8.12) 
+
+if (CMAKE_COMPILER_IS_GNUCC)
+  add_compile_options(-Wall -Wextra -Werror -Wfatal-errors)
```

Comment:
Please don't put `-Werror` in default compile options. It means new compiler versions often cause unnecessary fail-to-builds.

Also I'd leave `-Wfatal-errors` out, that one is personal preference (I tend to prefer getting multiple errors in one run)